src/tools/sssctl/sssctl_config.c

```diff
@@ -47,8 +78,15 @@ errno_t sssctl_config_check(struct sss_cmdline *cmdline,
     size_t num_ra_error, num_ra_success;
     char **strs = NULL;
     TALLOC_CTX *tmp_ctx = NULL;
-
-    ret = sss_tool_popt(cmdline, NULL, SSS_TOOL_OPT_OPTIONAL, NULL, NULL);
+    const char *config_path = NULL;
+    const char *config_snippet_path = NULL;
+    struct poptOption long_options[] = {
+        {"config", 'c', POPT_ARG_STRING, &config_path,
+            0, _("Specify a non-default config file"), NULL},
+        POPT_TABLEEND
+    };
+
+    ret = sss_tool_popt(cmdline, long_options, SSS_TOOL_OPT_OPTIONAL, NULL, NULL);
```

Comment:
Correct me if I am wrong:
In original version we were checking existence of any SSS_TOOL_OPT_OPTIONAL related config.
Now we are narrowing this checking to "-c" option only?
Shouldn't we first check all SSS_TOOL_OPT_OPTIONAL options as we did in the past and then evaluate on "-c" option?

Reply:
@elkoniu I think you get it wrong. Original config-check doesn't have any options. Here we extend it of one option -c/--config. Common options like --debug still works thanks to SSS_TOOL_OPT_OPTIONAL in the call.

The sss_tool_popt checks both options (common and command-specific) in one step.